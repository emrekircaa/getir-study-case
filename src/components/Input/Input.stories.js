import { Input } from "./index";

const meta = {
  title: "Components/Input",
  component: Input,
  parameters: {
    layout: "centered",
  },
  tags: ["autodocs"],
};

export default meta;

export const SampleInput = {
  args: {
    value: "value",
    placeHolder: "placeholder",
  },
};
