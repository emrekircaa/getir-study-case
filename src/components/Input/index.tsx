import React, { memo } from "react";
import styled from "styled-components";
import { colors } from "../../theme";

interface InputProps {
  placeHolder: string;
  value?: string;
  onChange?: (e: React.ChangeEvent<HTMLInputElement>) => void;
}

export const Input: React.FC<InputProps> = memo(
  ({ placeHolder, value, onChange }) => {
    return (
      <StyledInput
        data-testid="input"
        placeholder={placeHolder}
        value={value}
        onChange={onChange}
      />
    );
  }
);

const StyledInput = styled.input`
  border: 2px solid #e0e0e0;
  border-radius: 2px;
  padding: 12px 16px;
  height: 48px;
  outline: none;
  width: 100%;
  font-family: "Inter", sans-serif;

  &::placeholder {
    font-family: "Inter", sans-serif;
    font-size: 14px;
    line-height: 24px;
    letter-spacing: 0.15px;
  }
`;
