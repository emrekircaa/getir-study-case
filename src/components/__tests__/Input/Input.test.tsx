import React from "react";
import { render, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";
import { Input } from "../../Input/index";

describe("Input", () => {
  it("renders with correct props", () => {
    const { getByTestId } = render(
      <Input onChange={jest.fn()} placeHolder="placeholder" value="" />
    );
    const comp = getByTestId("input");
    expect(comp).toMatchSnapshot();
  });
  it("should match snapshot with value", () => {
    const { getByTestId } = render(
      <Input onChange={jest.fn()} placeHolder="placeholder" value="" />
    );
    const comp = getByTestId("input");
    expect(comp).toMatchSnapshot();
  });
  it("should trigger given onChange function on change", () => {
    const onChange = jest.fn();

    const { getByTestId } = render(
      <Input onChange={onChange} placeHolder="placeholder" value="" />
    );
    fireEvent.change(getByTestId("input"), { target: { value: "changed" } });
    expect(onChange).toBeCalledTimes(1);
  });
});
