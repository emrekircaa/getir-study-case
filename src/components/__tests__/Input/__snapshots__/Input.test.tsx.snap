// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders with correct props 1`] = `
<input
  class="sc-beySPh eZPbJQ"
  data-testid="input"
  placeholder="placeholder"
  value=""
/>
`;

exports[`Input should match snapshot with value 1`] = `
<input
  class="sc-beySPh eZPbJQ"
  data-testid="input"
  placeholder="placeholder"
  value=""
/>
`;
